### PR TITLE
Large cleanup of unnecessary and unused TPCC code

### PR DESCRIPTION
**Pull request by robertsami**
* Remove a ton of dead code
* Replace custom utilities with already included library utilities
* Fix a bunch of IntelliJ generated warnings

### CODE REVIEW

File: src/com/oltpbenchmark/LatencyRecord.java [hunk 1]
```diff
@@ -29,17 +29,15 @@
...
 
   public LatencyRecord(long startNs) {
     assert startNs > 0;
     allocateChunk();
   }
 
-    public void addLatency(int transType, long startNs, long endNs,
-                           long operationStartNs, long operationEndNs,
-                           int workerId, int phaseId) {
+  public void addLatency(int transType, long startNs, long endNs, long operationStartNs, long operationEndNs) {
```
Comment (psudheer21) [inline src/com/oltpbenchmark/LatencyRecord.java:40]: we need the operation start and end as well as the start and end iwth the wait times..
You can skip on the phase and the worker ids

File: src/com/oltpbenchmark/WorkloadConfiguration.java [hunk 2]
```diff
@@ -377,10 +317,6 @@ public boolean getUseThinkTime() {
     return useThinkTime;
   }
 
-  public void setUseThinkTime(boolean useThinkTime) {
```
Comment (psudheer21) [inline src/com/oltpbenchmark/WorkloadConfiguration.java]: is this not used?

File: src/com/oltpbenchmark/WorkloadConfiguration.java [hunk 3]
```diff
@@ -396,13 +332,6 @@ public void setShouldEnableForeignKeys(boolean shouldEnableForeignKeys) {
     this.shouldEnableForeignKeys = shouldEnableForeignKeys;
   }
 
-  public boolean getCreateSQLProcedures() {
```
Comment (psudheer21) [inline src/com/oltpbenchmark/WorkloadConfiguration.java:400]: is this not used?

File: src/com/oltpbenchmark/LatencyRecord.java (hunk 1 above)
Comment (robertsami) [reply src/com/oltpbenchmark/LatencyRecord.java:40]: what's the suggestion here?

File: src/com/oltpbenchmark/WorkloadConfiguration.java (hunk 2 above)
Comment (robertsami) [reply src/com/oltpbenchmark/WorkloadConfiguration.java]: correct

https://github.com/yugabyte/tpcc/search?q=setUseThinkTime

File: src/com/oltpbenchmark/WorkloadConfiguration.java (hunk 3 above)
Comment (robertsami) [reply src/com/oltpbenchmark/WorkloadConfiguration.java:400]: correct

https://github.com/yugabyte/tpcc/search?q=getCreateSQLProcedures

File: src/com/oltpbenchmark/LatencyRecord.java (hunk 1 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/LatencyRecord.java:40]: to keep this signature as is.
the 2 sets of start/end times are used here for different reasons.
The first one is for the entire operation including the wait times and the second one is just the latency of the operation.

File: src/com/oltpbenchmark/WorkloadConfiguration.java (hunk 2 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/WorkloadConfiguration.java]: Keep it for now. Not using this is a bug.
WIll correct it.

File: src/com/oltpbenchmark/WorkloadConfiguration.java (hunk 3 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/WorkloadConfiguration.java:400]: OK

File: src/com/oltpbenchmark/api/BenchmarkModule.java [hunk 4]
```diff
@@ -71,35 +64,22 @@
...
-     * Supplemental Procedures
-     */
-    private final Set<Class<? extends Procedure>> supplementalProcedures = new HashSet<Class<? extends Procedure>>();
-
     /**
      * A single Random object that should be re-used by all a benchmark's components
      */
     private final Random rng = new Random();
 
-    public BenchmarkModule(String benchmarkName, WorkloadConfiguration workConf, boolean withCatalog) throws Exception {
```
Comment (psudheer21) [inline src/com/oltpbenchmark/api/BenchmarkModule.java:94]: Did you try compiling with an older version to see if removing the "throws Exception" works with JAVA 1.8?

File: src/com/oltpbenchmark/api/BenchmarkModule.java [hunk 5]
```diff
@@ -452,11 +348,17 @@ public final TransactionType initTransactionType(String procName, int id) {
...
+        Map<String, Class<? extends Procedure>> txnClasses = Stream
+            .of(Delivery.class, NewOrder.class, OrderStatus.class, Payment.class, StockLevel.class)
+            .collect(Collectors.toMap(Class::getCanonicalName, clazz -> clazz));
+
         Package pkg = this.getProcedurePackageImpl();
         assert (pkg != null) : "Null Procedure package for " + this.benchmarkName;
         String fullName = pkg.getName() + "." + procName;
-        Class<? extends Procedure> procClass = (Class<? extends Procedure>) ClassUtil.getClass(fullName);
+
+        Class<? extends Procedure> procClass = txnClasses.get(fullName);
```
Comment (psudheer21) [inline src/com/oltpbenchmark/api/BenchmarkModule.java:359]: Why is this done?

File: src/com/oltpbenchmark/api/BenchmarkModule.java [hunk 6]
```diff
@@ -466,30 +368,16 @@ public final WorkloadConfiguration getWorkloadConfiguration() {
...
-                    txn = new TransactionType(procClass, procClass.hashCode(), true);
-                    txns.add(txn);
-                }
-            } // FOR
-
             for (TransactionType txn : txns) {
-                Procedure proc = (Procedure)ClassUtil.newInstance(txn.getProcedureClass(),
-                        new Object[0],
-                        new Class<?>[0]);
-                proc.initialize(this.workConf.getDBType());
```
Comment (psudheer21) [inline src/com/oltpbenchmark/api/BenchmarkModule.java:490]: NIce

File: src/com/oltpbenchmark/api/Worker.java [hunk 7]
```diff
@@ -431,36 +399,36 @@ public final void run() {
                     // that either started during the warmup phase or ended
                     // after the timer went off.
                     if (preState == State.MEASURE && executionState.getTransactionType() != null &&
-                        this.wrkldState.getCurrentPhase().id == phase.id) {
+                        Worker.wrkldState.getCurrentPhase().id == phase.id) {
```
Comment (psudheer21) [inline src/com/oltpbenchmark/api/Worker.java:417]: WHy is this change needed?

File: src/com/oltpbenchmark/api/Worker.java [hunk 8]
```diff
@@ -431,36 +399,36 @@ public final void run() {
...
                         latencies.addLatency(
                           executionState.getTransactionType().getId(),
                           start, end,
-                          executionState.getStartOperation(), executionState.getEndOperation(),
-                          this.id, phase.id);
+                          executionState.getStartOperation(), executionState.getEndOperation()
```
Comment (psudheer21) [inline src/com/oltpbenchmark/api/Worker.java:421]: REtain the old version.

File: src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java [hunk 9]
```diff
@@ -266,19 +255,13 @@ private void newOrderTransaction(int w_id, int d_id, int c_id,
       if (!rs.next())
         throw new RuntimeException("C_D_ID=" + d_id
             + " C_ID=" + c_id + " not found!");
-      c_discount = rs.getFloat("C_DISCOUNT");
```
Comment (psudheer21) [inline src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java]: The tpcc spec says that the values have to be stored somewhere.
I guess thats why they did it this way.

Retain all the unused varaibles as well. Let the compiler optimize it out if needed.

File: src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java [hunk 10]
```diff
@@ -673,7 +631,6 @@ int insertOrderLines(int o_id, int w_id, int d_id,
       stmtInsertOrderLine.setString(k++, ol_dist_info);
     }
     stmtInsertOrderLine.execute();
-    return total_amount;
```
Comment (psudheer21) [inline src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java]: Retain this as well.

File: src/com/oltpbenchmark/types/DatabaseType.java [hunk 11]
```diff
@@ -34,97 +34,46 @@
...
+    SQLAZURE(true),
+    ASSCLOWN(true),
+    HSQLDB(true),
+    H2(true),
+    MONETDB(true),
+    NUODB(true),
+    TIMESTEN(true),
+    CASSANDRA(false),
+    MEMSQL(false),
+    NOISEPAGE(true),
```
Comment (psudheer21) [inline src/com/oltpbenchmark/types/DatabaseType.java:54]: we can probably remove all except postgres.
Maybe we don't need this class itself?

File: src/com/oltpbenchmark/util/ScriptRunner.java [hunk 12]
```diff
@@ -24,7 +24,6 @@
 import java.io.InputStreamReader;
```
Comment (psudheer21) [inline src/com/oltpbenchmark/util/ScriptRunner.java:24]: Maybe we remove this as well?
And create tables the same way we load the tables using the connection object?

But thats for a different time. :D

File: src/com/oltpbenchmark/api/Worker.java (hunk 7 above)
Comment (robertsami) [reply src/com/oltpbenchmark/api/Worker.java:417]: its not strictly needed, but its more semantic java to refer to static members this way

File: src/com/oltpbenchmark/util/ScriptRunner.java (hunk 12 above)
Comment (robertsami) [reply src/com/oltpbenchmark/util/ScriptRunner.java:24]: agreed, though!

File: src/com/oltpbenchmark/LatencyRecord.java (hunk 1 above)
Comment (robertsami) [reply src/com/oltpbenchmark/LatencyRecord.java:40]: current signature is sufficient as discussed offline

File: src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java (hunk 9 above)
Comment (robertsami) [reply src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java]: done, in all procedures classes

File: src/com/oltpbenchmark/types/DatabaseType.java (hunk 11 above)
Comment (robertsami) [reply src/com/oltpbenchmark/types/DatabaseType.java:54]: i agree, but id prefer to make further removals/deletions in a follow up at this point, given the already massive scope of this change

File: src/com/oltpbenchmark/api/Worker.java (hunk 8 above)
Comment (robertsami) [reply src/com/oltpbenchmark/api/Worker.java:421]: resolving per same reason in associated comment on function def'n

File: src/com/oltpbenchmark/api/BenchmarkModule.java (hunk 5 above)
Comment (robertsami) [reply src/com/oltpbenchmark/api/BenchmarkModule.java:359]: So we can remove ClassUtil

**Review by psudheer21 — APPROVED** [review]
